main/vm.json

```diff
@@ -0,0 +1,50 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "location": {
+      "type": "string"
+    },
+    "vmName": {
+      "type": "string"
+    },
+    "hardwareProfile": {
+      "type": "object"
+    },
+    "osProfile": {
+      "type": "object"
+    },
+    "storageProfile": {
+      "type": "object"
+    },
+    "networkProfile": {
+      "type": "object"
+    },
+    "diagnosticsProfile": {
+      "type": "object"
+    },
+    "plan": {
+      "type": "object"
+    }
+  },
+  "variables": {
+    "properties": {
+      "hardwareProfile": "[parameters('hardwareProfile')]",
+      "osProfile": "[parameters('osProfile')]",
+      "storageProfile": "[parameters('storageProfile')]",
+      "networkProfile": "[parameters('networkProfile')]",
+      "diagnosticsProfile": "[parameters('diagnosticsProfile')]"
+    }
+  },
+  "resources": [
+    {
+      "apiVersion": "2020-12-01",
```

Comment:
why this date?

Reply:
This is format of Azure.

Reply:
Format is fine why it is 2020? Do you copied from NIOS template?

Reply:
Updated all apiVersion fields to latest.